src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntTests.cs

```diff
@@ -1454,41 +1454,211 @@ .maxstack  1
         // PROTOTYPE: Test unary operator- with `static IntPtr operator-(IntPtr)` defined on System.IntPtr. (Should be ignored for `nint`.)
         public static IEnumerable<object[]> UnaryOperatorsData()
```

Comment:
>UnaryOperatorsData [](start = 44, length = 18)

nit: what's the advantage of extracting this "data production" method, given that it's only used in one test?

I'd be tempted to invert the pattern and avoid this `object[]` serialization entirely.
The test could do:
```
void Test()
{
  validate(... combination1 ...)
  validate(... combination2 ...)
  validate(... combination3 ...)

  void validate(....)
  {
    ...
  }
}
```

Reply:
Done.

---
In reply to: [373736408](https://github.com/dotnet/roslyn/pull/41030#discussion_r373736408) [](ancestors = 373736408)